pallet/deposit/src/lib.rs

```diff
@@ -169,6 +164,16 @@ pub mod pallet {
 			}
 
 			<Deposits<T>>::try_mutate(&who, |ds| {
+				let ds = if let Some(ds) = ds {
```

Comment:
Suggest `let ds = if let Some(ds) = maybe_bs {`

Reply:
I prefer to shadow the variable if we don't need to access it later.